Fix musl build

Inspired by http://patchwork.openembedded.org/patch/111661/
"inb/outb are implemented for ARM on glibc but not on linux in general
therefore the conditional has to reflect that"

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr libpciaccess-0.13.4.org/src/linux_sysfs.c libpciaccess-0.13.4/src/linux_sysfs.c
--- libpciaccess-0.13.4.org/src/linux_sysfs.c	2015-05-01 06:44:47.000000000 +0200
+++ libpciaccess-0.13.4/src/linux_sysfs.c	2016-01-24 15:27:57.141562678 +0100
@@ -49,7 +49,7 @@
 #include <dirent.h>
 #include <errno.h>
 
-#if defined(__i386__) || defined(__x86_64__) || defined(__arm__)
+#if defined(__i386__) || defined(__x86_64__)
 #include <sys/io.h>
 #else
 #define inb(x) -1
